src/client/src/ui/common/popout/reactPopoutService.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { PopoutOptions } from './';
+import './popoutRegion.scss';
+
+let POPOUT_CONTAINER_ID = 'container_id';
+
+export default class ReactPopoutService {
+  openPopout(options:PopoutOptions, view:React.Component) {
+    let popoutContainer;
+    let windowOptionsString = this._getWindowOptionsString(options.windowOptions);
+    let childWindow = window.open(options.url, options.title, windowOptionsString);
+    let onloadHandler = () => {
+      childWindow.document.title = options.title;
+      popoutContainer = childWindow.document.createElement('div');
+      popoutContainer.id = POPOUT_CONTAINER_ID;
+      childWindow.document.body.appendChild(popoutContainer);
```

Comment:
maybe we should think about passing a selector for rootNode and document.body as backup. this allows you to have a proper dom present and an element where to install vs `document.body` only.

also, it's possible onload has already fired by the time the fn runs (likely with an empty doc), should check readystate or assign onload handlers.